src/plugins/visualizations/public/visualize_app/utils/get_visualization_instance.ts

```diff
@@ -47,6 +48,7 @@ const createVisualizeEmbeddableAndLinkSavedSearch = async (
         search: data.search,
         savedObjectsClient: savedObjects.client,
         spaces,
+        savedObjectsTagging,
```

Comment:
Does tagging have any use in the context of a visualization? If not, do we really to pull this dependency into the visualizations plugin? It appears to be an optional argument to `getSavedSearch`.

Reply:
Not that I'm aware of. If you'd prefer I can remove the dependency and pass `undefined` instead.

Reply:
Let's keep it so that no one is confused in the future by an empty tags array.

It seems odd to me that `getSavedSearch` can be called with or without the tags dependency and there's no way to know from the result whether tags were actually loaded or not because the tags array is present either way. Is it empty because the saved search has no tags or because a previous developer didn't provide `savedObjectsTagging`?

But, this is very minor. It's also your team's code and you know more about it than I do, so do whatever seems best to you.

Thanks for working on this!

Reply:
> It seems odd to me that `getSavedSearch` can be called with or without the tags dependency and there's no way to know from the result whether tags were actually loaded or not because the tags array is present either way. Is it empty because the saved search has no tags or because a previous developer didn't provide `savedObjectsTagging`?

That's a good point. For the work in this PR specifically it's not important, but the distinction between a search not having tags and the tags not being fetched could matter in some situations. I don't think we can guarantee `savedObjectsTaggingOss.getTaggingApi()` won't return `undefined` due to its dependency on `savedObjectsTagging`, but then maybe it makes sense to have the tags property be `string[] | undefined` or similar.

I'm going to unresolve this and tag @kertal to see what he thinks, and I'll re-resolve once we've decided. Thanks!

Reply:
+1 for `string[] | undefined`, think this makes sense in this case

Reply:
+1

Reply:
`string[] | undefined` it is!